### PR TITLE
[bitnami/elasticsearch] allow existing claims in master and data

### PR DESCRIPTION
**Pull request by zenbones**
For bitnami/elasticsearch...

This change allows master and data to use an existing volume claim for the data volume, as opposed to a claim template. Many other bitnami charts allow existing volume claims in this way, which are a convenience for the user.

This change, especially in light of stateful sets, means that the user does have to be responsible for the pvc, and all that entails. But, by using the existingClaim value, that is exactly what they're asking for. They can always get the previous behavior merely by not including the value, so I don't think there's any downside.

### CODE REVIEW

File: bitnami/elasticsearch/templates/data-statefulset.yaml [hunk 1]
```diff
@@ -216,18 +216,22 @@ spec:
...
+        {{- end }}        
       spec:
         accessModes: {{- toYaml .Values.data.persistence.accessModes | nindent 10 }}
         {{ $storage := dict "global" .Values.global "local" .Values.data }}
         {{ include "elasticsearch.storageClass" $storage }}
         resources:
           requests:
             storage: {{ .Values.data.persistence.size | quote }}
-{{- end }}
+{{- end }}
```
Comment (marcosbc) [inline bitnami/elasticsearch/templates/data-statefulset.yaml:237]: Could you fix the "No newline at end of file" error?

File: bitnami/elasticsearch/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: elasticsearch
-version: 12.3.6
+version: 12.3.7
```
Comment (marcosbc) [inline bitnami/elasticsearch/Chart.yaml:3]: This would be in any case a minor version change (i.e. 12.4.0)

**Comment by marcosbc** [comment]
Thanks for the contribution! I added a few comments on a few improvements, let me know if you have any question.

File: bitnami/elasticsearch/templates/data-statefulset.yaml (hunk 1 above)
Comment (zenbones) [reply bitnami/elasticsearch/templates/data-statefulset.yaml:237]: Passed the linter, so I'm not sure where this error is. I'll add an extraneous new line to both files I changed. Maybe that will solve it?

Comment (zenbones) [reply bitnami/elasticsearch/templates/data-statefulset.yaml]: fixed

File: bitnami/elasticsearch/Chart.yaml (hunk 2 above)
Comment (zenbones) [reply bitnami/elasticsearch/Chart.yaml]: changed to 12.4.0

**Comment by zenbones** [comment]
Did you want me to take a pass at the values.yaml, values-production.yaml and README?

**Comment by zenbones** [comment]
Closed, the conversation... not sure that was right, but in case yo can't see them, I made the suggested changes.

File: bitnami/elasticsearch/templates/data-statefulset.yaml [hunk 3]
```diff
@@ -216,13 +216,17 @@ spec:
...
+          persistentVolumeClaim:
+            claimName: {{ .Values.data.persistence.existingClaim }}
 {{- else }}
   volumeClaimTemplates:
     - metadata:
         name: "data"
         {{- if .Values.data.persistence.annotations }}
         annotations: {{- toYaml .Values.data.persistence.annotations | nindent 10 }}
-        {{- end }}
+        {{- end }}
```
Comment (marcosbc) [inline bitnami/elasticsearch/templates/data-statefulset.yaml:229]: It looks like some trailing spaces were added, could you remove them?

Comment (zenbones) [reply bitnami/elasticsearch/templates/data-statefulset.yaml]: removed trailing spaces, added documentation to values.yaml, values-production.yaml and README.md

**Comment by zenbones** [comment]
Let me know if you want any other changes.

**Pull request merged**